src/error.ts

```diff
@@ -14,110 +14,109 @@ export enum codes {
 }
 
 export class InappropriateMultiaddrError extends WebRTCTransportError {
   constructor (msg: string) {
-    super(`There was a problem with the Multiaddr which was passed in: ${msg}`)
+    super(`There was a problem with the Multiaddr which was passed in: ${msg}`, codes.ERR_INVALID_MULTIADDR)
     this.name = 'WebRTC/InappropriateMultiaddrError'
   }
 }
 
 export function inappropriateMultiaddr (msg: string) {
-  return errCode(new InappropriateMultiaddrError(msg), codes.ERR_INVALID_MULTIADDR)
+  return new InappropriateMultiaddrError(msg)
 }
 
 export class InvalidArgumentError extends WebRTCTransportError {
   constructor (msg: string) {
-    super(`There was a problem with a provided argument: ${msg}`)
+    super(`There was a problem with a provided argument: ${msg}`, codes.ERR_INVALID_PARAMETERS)
     this.name = 'WebRTC/InvalidArgumentError'
   }
 }
 
 export function invalidArgument (msg: string) {
-  return errCode(new InvalidArgumentError(msg), codes.ERR_INVALID_PARAMETERS)
+  return new InvalidArgumentError(msg)
 }
 
 export class InvalidFingerprintError extends WebRTCTransportError {
   constructor (fingerprint: string, source: string) {
-    super(`Invalid fingerprint "${fingerprint}" within ${source}`)
+    super(`Invalid fingerprint "${fingerprint}" within ${source}`, codes.ERR_INVALID_FINGERPRINT)
     this.name = 'WebRTC/InvalidFingerprintError'
   }
 }
 
 export function invalidFingerprint (fingerprint: string, source: string) {
-  return errCode(new InvalidFingerprintError(fingerprint, source), codes.ERR_INVALID_FINGERPRINT)
+  return new InvalidFingerprintError(fingerprint, source)
 }
 
 export class OperationAbortedError extends WebRTCTransportError {
   constructor (context: string, abortReason: string) {
-    super(`Signalled to abort because (${abortReason}}) ${context}`)
+    super(`Signalled to abort because (${abortReason}}) ${context}`, codes.ERR_ALREADY_ABORTED)
     this.name = 'WebRTC/OperationAbortedError'
   }
 }
 
 export function operationAborted (context: string, reason: string) {
-  return errCode(new OperationAbortedError(context, reason), codes.ERR_ALREADY_ABORTED)
+  return new OperationAbortedError(context, reason)
 }
 
 export class OverStreamLimitError extends WebRTCTransportError {
-  constructor (msg: string) {
-    super(msg)
+  constructor (dir: Direction, proto: string) {
+    const code = dir === 'inbound' ? codes.ERR_TOO_MANY_INBOUND_PROTOCOL_STREAMS : codes.ERR_TOO_MANY_OUTBOUND_PROTOCOL_STREAMS
+    super(`${dir} stream limit reached for protocol - ${proto}`, code)
     this.name = 'WebRTC/OverStreamLimitError'
   }
 }
 
 export function overStreamLimit (dir: Direction, proto: string) {
-  const code = dir === 'inbound' ? codes.ERR_TOO_MANY_INBOUND_PROTOCOL_STREAMS : codes.ERR_TOO_MANY_OUTBOUND_PROTOCOL_STREAMS
-  return errCode(new OverStreamLimitError(`${dir} stream limit reached for protocol - ${proto}`), code)
+  return new OverStreamLimitError(dir, proto)
 }
```

Comment:
Breaking change has to do with exported `OverStreamLimitError` parameter change.

Reply:
Same comment here, we could have a default value for `proto` param in the `OverStreamLimitError` constructor (e.g. `unknown`).  We could add TODOs and make the breaking changes with a planned major version increase.

Reply:
I can make proto in `overStreamLimit` optional; note this would be adding a change.
The parameter change I made has to do with the `OverStreamLimitError` which completely changes the parameters.
I'll undo the breaking parts and add a note.